### PR TITLE
Content Host fixtures updated to filter tests based on workflow - dynamically

### PR DESCRIPTION
**Pull request by jyejare**
- Content hosts fixtures are updated to prioritize content_host conf workflow setting, allowing pytest to filter tests based on Workflows like `pit_wf` or `fips_wf`.
- This update eliminates the use of `broker_settings.yaml` nicks as we are directly providing the nick parameters to the VMBroker Context manager.
- The tests can be filtered with wf_name using -k `wf_name` to pytest as content hosts tests_names are appended with wf_name using pytest ids.

Collection:
The Pytest Collection to filter the tests based on workflow (e.g`pit_wf`) set in `content_host.yaml`:

```
% pytest -m 'content_host' --collect-only tests/foreman/ 
========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.8.2, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/jitendrayejare/JWorkspace/GitRepos/robottelo, configfile: pyproject.toml
plugins: cov-2.12.1, ibutsu-1.16, reportportal-5.0.8, xdist-2.3.0, services-2.2.1, mock-3.6.1, forked-1.3.0
collected 5460 items / 5342 deselected / 118 selected                                                                                                                                                    

<Package api>
  <Module test_contentmanagement.py>
    <Class TestCapsuleContentManagement>
        <Function test_positive_mirror_on_sync[rhel7_contenthost0]>
  <Module test_errata.py>
    <Function test_positive_install_in_host[rhel7_contenthost0]>
    <Function test_positive_install_multiple_in_host[rhel7_contenthost0]>
    <Function test_positive_get_count_for_host[rhel6_contenthost0]>
    <Function test_positive_get_applicable_for_host[rhel6_contenthost0]>
    <Function test_positive_incremental_update_required[rhel7_contenthost0]>
    <Function test_errata_installation_with_swidtags[rhel8_contenthost0]>
    <Function test_apply_modular_errata_using_default_content_view[rhel8_contenthost0]>
  <Module test_subscription.py>
    <Function test_positive_subscription_status_disabled[rhel_6]>
    <Function test_positive_subscription_status_disabled[rhel_7]>
    <Function test_positive_subscription_status_disabled[rhel_8]>
.
.
.
.
=============== 118/5460 tests collected (5342 deselected) in 25.98s ============
```

### CODE REVIEW

File: conf/broker.yaml.template [hunk 1]
```diff
@@ -4,3 +4,17 @@ BROKER:
     HOST_WORKFLOWS:
         POWER_CONTROL: vm-power-operation
         EXTEND: extend-vm
+    # Client release versions and confs
+    CLIENTS:
```
Comment (mshriver) [inline conf/broker.yaml.template:8]: I think this config should go in `content_host.yaml`, as that's where we should be defining versions, workflows, and other settings data concerning content hosts.

File: pytest_fixtures/content_hosts.py [hunk 2]
```diff
@@ -26,52 +38,113 @@ def rhel_contenthost(request):
...
+        host_classes={'host': ContentHost},
+    ) as host:
         yield host
 
 
-@pytest.fixture(scope="module")
-def rhel7_contenthost_module():
+@pytest.fixture(
+    scope='module',
+    params=[(settings.content_host.deploy_workflow, '7')],
```
Comment (bherrin3) [inline pytest_fixtures/content_hosts.py:73]: Bad idea. These can change on reconfigure of AT and DEFINITELY change whenever we re-deploy... the WF order for configuration changes (like if I re-order them in YAML and redeploy).

File: pytest_fixtures/content_hosts.py [hunk 3]
```diff
@@ -26,52 +38,113 @@ def rhel_contenthost(request):
...
+        cores=host_conf['cores'],
+        host_classes={'host': ContentHost},
+    ) as host:
         yield host
 
 
-@pytest.fixture
-def rhel8_contenthost():
+@pytest.fixture(
+    params=[(settings.content_host.deploy_workflow, '8')],
```
Comment (bherrin3) [inline pytest_fixtures/content_hosts.py:89]: Same bad idea.

File: pytest_fixtures/content_hosts.py (hunk 2 above)
Comment (mshriver) [reply pytest_fixtures/content_hosts.py:73]: its not, 7 is the rhel major version.

**Comment by jyejare** [comment]
@mshriver Please revisit again, I removed the necessity of `-k` to filter content host tests now!

File: pytest_fixtures/content_hosts.py [hunk 4]
```diff
@@ -8,99 +8,78 @@
...
+    """A fixture for use with unittest classes. Provides a Content Host object"""
+    with VMBroker(**host_conf, host_classes={'host': ContentHost}) as host:
         yield host
 
 
-@pytest.fixture(scope="module")
-def rhel7_contenthost_module():
+@pytest.fixture(scope='module', params=[{'rhel_version': '7'}])
+def rhel7_contenthost_module(host_conf):
     """A module-level fixture that provides a content host object based on the rhel7 nick"""
```
Comment (omkarkhatavkar) [inline pytest_fixtures/content_hosts.py:53]: this is no more using `nick`, update as host object as you mentioned for other doc string

File: pytest_fixtures/content_hosts.py [hunk 5]
```diff
@@ -8,99 +8,78 @@
...
 
 @pytest.fixture
-def rhel_contenthost(request):
+def host_conf(request):
+    """A function-level fixture that returns parameters for VMBroker host deployment"""
+    conf = params = {}
+    if hasattr(request.node, 'callspec'):
+        params = [*request.node.callspec.params.values()][0]
+    conf['workflow'] = params.get('workflow', settings.content_host.deploy_workflow)
+    _rhelver = f"rhel{params.get('rhel_version', settings.content_host.default_rhel_version)}"
```
Comment (omkarkhatavkar) [inline pytest_fixtures/content_hosts.py:21]: single quote

File: pytest_fixtures/content_hosts.py [hunk 6]
```diff
@@ -8,99 +8,78 @@
...
+    conf['rhel_ver'] = settings.content_host.hardware.get(f'{_rhelver}').release
```
Comment (omkarkhatavkar) [inline pytest_fixtures/content_hosts.py:22]: can't you directly access the `settings.content_host.hardware.get(_rhelver).release` as _rhelver is string object seems

File: pytest_fixtures/content_hosts.py (hunk 5 above)
Comment (jyejare) [reply pytest_fixtures/content_hosts.py:21]: nop, there is a `rhel_version` already has a single quote hence I used double quotes outside.

**Review by omkarkhatavkar — APPROVED** [review]
ACK

File: conf/broker.yaml.template (hunk 1 above)
Comment (jyejare) [reply conf/broker.yaml.template]: Moved to `content_host.yaml`.

**Pull request merged**